Libraries/CommonFunctions.psm1

```diff
@@ -4284,4 +4284,106 @@ Function Remove-InvalidCharactersFromFileName
     $WindowsInvalidCharacters = [IO.Path]::GetInvalidFileNameChars() -join ''
     $Regex = "[{0}]" -f [RegEx]::Escape($WindowsInvalidCharacters)
     return ($FileName -replace $Regex)
+}
+
+function Get-KVPItem {
+	param (
+	    $VMName,
+	    $server,
+	    $keyName,
+		$Intrinsic
+	)
+
+    $vm = Get-WmiObject -Namespace root\virtualization\v2 -Query "Select * From Msvm_ComputerSystem Where ElementName=`'$VMName`'"
+    if (-not $vm)
+    {
+        return $Null
+    }
+
+    $kvpEc = Get-WmiObject -Namespace root\virtualization\v2 -Query "Associators of {$vm} Where AssocClass=Msvm_SystemDevice ResultClass=Msvm_KvpExchangeComponent"
+    if (-not $kvpEc)
+    {
+        return $Null
+    }
+
+    $kvpData = $Null
+
+    if ($Intrinsic)
+    {
+        $kvpData = $KvpEc.GuestIntrinsicExchangeItems
+    }
+    else
+    {
+        $kvpData = $KvpEc.GuestExchangeItems
+    }
+
+    if ($kvpData -eq $Null)
+    {
+        return $Null
+    }
+
+    foreach ($dataItem in $kvpData)
+    {
+        $key = $null
+        $value = $null
+        $xmlData = [Xml] $dataItem
+
+        foreach ($p in $xmlData.INSTANCE.PROPERTY)
+        {
+            if ($p.Name -eq "Name")
+            {
+                $key = $p.Value
+            }
+
+            if ($p.Name -eq "Data")
+            {
+                $value = $p.Value
+            }
+        }
+        if ($key -eq $keyName)
+        {
+            return $value
+        }
+    }
+
+    return $Null
+}
+
+function Add-RemoveMaxNIC {
+	param(
+		$vmName,
+		$hvServer,
+		$network_type,
+		$actionType,
+		[int] $nicsAmount
+	)
+	for ($i=1; $i -le $nicsAmount; $i++)
+	{
+		$nicName = "External" + $i
+
+		if ($actionType -eq "add")
+		{
+			"Info : Ensure the VM does not have a Synthetic NIC with the name '${nicName}'"

```

Comment:
LogMsg "Ensure ... " (use LogMsg and remove "Info : " from the message). There are other occurrences

Reply:
Changed